bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,221 @@
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @ValueMapValue(name = ExperienceFragment.PN_FRAGMENT_PATH, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String fragmentPath;
+
+    @OSGiService
+    private LanguageManager languageManager;
+
+    @OSGiService
+    private LiveRelationshipManager relationshipManager;
+
+    private String localizedFragmentPath;
+
+    @PostConstruct
+    protected void initModel() {
+        if (inTemplate()) {
+            String currentPageRootPath = getRoot(currentPage.getPath());
+            // we should use getRoot instead of getXfRoot once the XF UI supports creating Live and Language Copies
+            String xfRootPath = getXfRoot(fragmentPath, currentPageRootPath);
+            if (!StringUtils.isEmpty(currentPageRootPath) && !StringUtils.isEmpty(xfRootPath)) {
+                String xfRelativePath = StringUtils.substring(fragmentPath, xfRootPath.length());
+                String localizedXfRootPath = StringUtils.replace(currentPageRootPath, CONTENT_ROOT, EXPERIENCE_FRAGMENTS_ROOT, 1);
+                localizedFragmentPath = StringUtils.join(localizedXfRootPath, xfRelativePath, JCR_CONTENT_ROOT);
+            }
+        }
+        String xfContentPath = StringUtils.join(fragmentPath, JCR_CONTENT_ROOT);
+        if (!resourceExists(localizedFragmentPath) && resourceExists(xfContentPath)) {
+            localizedFragmentPath = xfContentPath;
+        }
+    }
+
+    @Override
+    public String getFragmentPath() {
+        return fragmentPath;
+    }
+
+    @Override
+    public String getLocalizedFragmentPath() {
+        return localizedFragmentPath;
+    }
+
+    @NotNull
+    @Override public String getExportedType() {
+        return request.getResource().getResourceType();
+    }
+
+    /**
+     * Returns the localization root of the resource defined at the given path
+     *
+     * Use case                                  | Path                                 | Root
+     * ----------------------------------------  |--------------------------------------|------------------
+     * 1. No localization                        | /content/mysite/mypage               | null
+     * 2. Language localization                  | /content/mysite/en/mypage            | /content/mysite/en
+     * 3. Country-language localization          | /content/mysite/us/en/mypage         | /content/mysite/us/en
+     * 4. Country-language localization (variant)| /content/us/mysite/en/mypage         | /content/us/mysite/en
+     * 5. Blueprint                              | /content/mysite/blueprint/mypage     | /content/us/mysite/blueprint
+     * 4. Livecopy                               | /content/mysite/livecopy/mypage      | /content/us/mysite/livecopy
+     *
+     * @param path the resource path.
+     * @return the localization root of the resource at the given path if it exists, {@code null} otherwise.
+     */
+    private String getRoot(String path) {
```

Comment:
you might consider extraction of logic related to path calculation to separate class. the class might be enhanced about unit/integration tests which cover all cases

Reply:
All cases should already be covered by JUnit tests.